### PR TITLE
Add pangolin v4

### PR DESCRIPTION
**Pull request by kapsakcj**
- [x] This comment contains a description of what is in the pull request.
This PR adds:
-  dockerfile for Pangolin v4.0
- GH actions workflow for testing
- GH actions workflow for deployment
- updated README.md for basic pangolin usage

Versions pinned in Dockerfile, used to create the conda environment
```
ARG PANGOLIN_VER="v4.0"
ARG PANGOLIN_DATA_VER="v1.2.133.2"
ARG SCORPIO_VER="v0.3.16"
ARG CONSTELLATIONS_VER="v0.1.4"
ARG USHER_VER="0.5.3"
```
Though this is not accurately reflected with `pangolin --all-versions`. It's a tad bit confusing but [more info can be found here](https://github.com/cov-lineages/pangolin/issues/393). I don't want to wait for a fix to be put in place so I'd like to deploy this asap and warn users about the discrepancy.
```
(pangolin) root@707c2228bd71:/data# pangolin --all-versions
****
Pangolin running in usher mode.
****
pangolin: 4.0
pangolin-data: 1.2.133
constellations: v0.1.4
scorpio: 0.3.16
pangolin-assignment: v1.2.133
```

- [x] Build your own docker image using a Dockerfile
  - [x] Directory structure should be name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. spades/3.12.0/Dockerfile)
  - [x] Includes the recommended LABELS
- [x] (Optional) Dockerfile is built with best practices and has been approved by a linter (such as https://hadolint.github.io/hadolint/)
- [x] Edit main README.md
- [x] Ensure tool is listed in Program_Licenses.md
- [x] Create a simple container-specific README.md in the same directory as the Dockerfile (i.e. spades/3.12.0/README.md)
- [ ] Update GitHub actions workflow if needed
- [x] Any files required for building are located in the same directory as the Dockerfile (i.e. spades/3.12.0/my_spades_tests.sh)
- [ ] Have successfully run the workflow "Test <program name> image" in your forked repository

### CODE REVIEW

**Comment by erinyoung** [comment]
I'm so excited for this!

**Comment by kapsakcj** [comment]
Hmm looks like the older GH actions workflow kicked off in addition to the one I just added.

 Guessing it's due to this line with a `*`: https://github.com/kapsakcj/docker-builds/blob/8b7e7770430cfb7b45e80ac556e40604715f600d/.github/workflows/test-pangolin3.1.20-slim.yml#L7

**Comment by kapsakcj** [comment]
I have a feeling that this error is related to RAM usage while testing pangolearn/fast mode. The tests work perfect on my VM with 8cpus and 30GB of RAM.

The 7GB on GH actions runners may not be enough https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

https://github.com/cov-lineages/pangolin/issues/395

**Comment by kapsakcj** [comment]
Yep, its the RAM usage.

When I ran `$ docker build -m 7000000000 --target test -t kapsakcj/pangolin:4.0 pangolin/4.0/` which limits RAM to 7GB, the test failed with the same error when attempting pangolearn analysis-mode.

Going to remove pangolearn testing from the test layer

**Comment by erinyoung** [comment]
Looks like the tests all worked. Can we merge and deploy this?

I tested it locally with some fastas, and it worked for me.

File: README.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ To learn more about the docker pull rate limits and the open source software pro
...
...021-04-28</li><li>2.4.2 & 2021-05-10</li><li>2.4.2 & 2021-05-11</li><li>2.4.2 & 2021-05-19</li><li>3.0.5 & 2021-06-05</li><li>3.1.3 & 2021-06-15</li><li>3.1.5 & 2021-06-15</li><li>3.1.5 & 2021-07-07-2</li><li>3.1.7 & 2021-07-09</li><li>3.1.8 & 2021-07-28</li><li>3.1.10 & 2021-07-28</li><li>3.1.11 & 2021-08-09</li><li>3.1.11 & 2021-08-24</li><li>3.1.11 & 2021-09-17</li><li>3.1.14 & 2021-09-28</li><li>3.1.14 & 2021-10-13</li><li>3.1.16 & 2021-10-18</li><li>3.1.16 & 2021-11-04</li><li>3.1.16 & 2021-11-09</li><li>3.1.16 & 2021-11-18</li><li>3.1.16 & 2021-11-25</li><li>3.1.17 & 2021-11-25</li><li>3.1.17 & 2021-12-06</li><li>3.1.17 & 2022-01-05</li><li>3.1.18 & 2022-01-20</li><li>3.1.19 & 2022-01-20</li><li>3.1.20 & 2022-02-02</li><li>3.1.20 & 2022-02-28</li></ul>**Pangolin version & pangolin-data version** </br><ul><li>4.0 & 1.2.133</li></ul> |  https://github.com/cov-lineages/pangolin<br/>https://github.com/cov-lineages/pangoLEARN<br/>https://github.com/cov-lineages/pango-designation<br/>https://github.com/cov-lineages/scorpio<br/>https://github.com/cov-lineages/constellations<br/>https://github.com/cov-lineages/lineages (archived)<br/>https://github.com/hCoV-2019/pangolin (archived) |
```
Comment (erinyoung) [inline README.md:95]: This is getting so long

**Comment by kapsakcj** [comment]
Sure, go for it!

**Review by erinyoung — APPROVED** [review]

**Pull request merged**